### PR TITLE
Use data providers in creators

### PR DESCRIPTION
**Pull request by thz**
**What this PR does / why we need it**:

EnsureRole (et al) shall be used in seed and user-cluster side. TemplateData struct is specific to seed side.
For that reason Ensure functions should use interfaces instead of TemplateData struct which can be used from both sides.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by thz** [comment]
@mrIncompetent like this?

**Comment by thz** [comment]
ping @mrIncompetent

**Comment by thz** [comment]
ping @mrIncompetent

File: api/pkg/resources/data.go [hunk 1]
```diff
@@ -19,23 +19,105 @@ import (
...
-	InClusterPrometheusDisableDefaultScrapingConfigs bool
-	InClusterPrometheusScrapingConfigsFile           string
+	nodePortRange                                    string
+	nodeAccessNetwork                                string
+	etcdDiskSize                                     resource.Quantity
+	inClusterPrometheusRulesFile                     string
+	inClusterPrometheusDisableDefaultRules           bool
+	inClusterPrometheusDisableDefaultScrapingConfigs bool
+	inClusterPrometheusScrapingConfigsFile           string
 	DockerPullConfigJSON                             []byte
```
Comment (mrIncompetent) [inline api/pkg/resources/data.go:36]: Whats the reason for some properties still being exported?

**Review by mrIncompetent — APPROVED** [review]

File: api/pkg/resources/data.go (hunk 1 above)
Comment (thz) [reply api/pkg/resources/data.go:36]: I was not aiming for complete conversion but only providing enough in the interfaces to allow the first migrations of ensure functions.

**Pull request merged**